server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoTileGridAggregatorTests.java

```diff
@@ -50,26 +52,49 @@ protected GeoBoundingBox randomBBox() {
                 );
             }
         );
-        // Avoid numerical errors for sub-atomic values
-        double left = GeoEncodingUtils.decodeLongitude(GeoEncodingUtils.encodeLongitude(bbox.left()));
-        double right = GeoEncodingUtils.decodeLongitude(GeoEncodingUtils.encodeLongitude(bbox.right()));
-        double top = GeoEncodingUtils.decodeLatitude(GeoEncodingUtils.encodeLatitude(bbox.top()));
-        double bottom = GeoEncodingUtils.decodeLatitude(GeoEncodingUtils.encodeLatitude(bbox.bottom()));
+        final int tiles = 1 << precision;
+
+        // Due to the way GeoTileBoundedPredicate works that adjust the given bounding box when it is touching the tiles, we need to
```

Comment:
I'm slightly concerned that somehow we might be hiding issues, but cannot really argue that case yet.